src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs

```diff
@@ -142,18 +144,18 @@ public void BeginGetGraphData(IGraphContext context)
 
             Assumes.True(IsInitialized);
 
-            GraphNodeId newNodeId = GetTopLevelGraphNodeId(projectPath, viewModel.OriginalModel.GetTopLevelId());
+            GraphNodeId newNodeId = GetTopLevelGraphNodeId(projectPath, viewModel.OriginalModel!.GetTopLevelId());
             return DoAddGraphNode(newNodeId, graphContext, projectPath, parentNode: null, viewModel);
         }
 
         private GraphNode DoAddGraphNode(
             GraphNodeId graphNodeId,
             IGraphContext graphContext,
             string projectPath,
-            GraphNode parentNode,
+            GraphNode? parentNode,
             IDependencyViewModel viewModel)
         {
-            _iconCache.Register(viewModel.Icon);
+            _iconCache!.Register(viewModel.Icon);
```

Comment:
I think for this pattern where the analysis doesn't know about `InitializeCoreAsync`, it might be nicer to define the field as `GraphIconCache _iconCache = null!;` rather than having to annotate every call site (or the first call site in every method)

Reply:
I debated this. In this class only one member uses the field, but I agree that in the general case this may be better.

Reply:
Yeah, this case was fine, but since you're (one of?) the first to annotate things we should probably establish some patterns.

Reply:
In the main I think it's a great addition to the language. Applying it was pretty fast and painless, with only a lessons learned, pointed to in my self-review comments. We definitely become dependent upon upstream types being annotated (corefx, CPS and other VS packages, ...).